### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by guyboertje**
Improve docs so users better understand json_lines and file input interop

### CODE REVIEW

File: lib/logstash/codecs/json_lines.rb [hunk 1]
```diff
@@ -23,30 +26,34 @@ class LogStash::Codecs::JSONLines < LogStash::Codecs::Base
...
     @on_event.call(event, event.to_json + NL)
   end # def encode
 
+  private
+
+  def guard(event, data)
+    begin
+      LogStash::Event.new(LogStash::Json.load(event["message"]))
+    rescue LogStash::Json::ParserError => e
+      @logger.info("JSON parse failure. Falling back to plain-text", :error => e, :data => data)
```
Comment (jsvd) [inline lib/logstash/codecs/json_lines.rb]: not sure if we want this as info, maybe debug?

One example of a similar scenario we have is grok, which doesn't log at all:
https://github.com/logstash-plugins/logstash-filter-grok/blob/master/lib/logstash/filters/grok.rb#L292-L301

Comment (guyboertje) [inline lib/logstash/codecs/json_lines.rb]: Thanks for the insight.  I just copy-paste'd.  I did/do not know enough of the whole stack behaviour to make this kind of value judgement. :-)

Comment (guyboertje) [inline lib/logstash/codecs/json_lines.rb]: @purbon @jsvd - what should I change it to? Log debug or no logging?

Comment (jsvd) [inline lib/logstash/codecs/json_lines.rb]: I don't mind not logging as tagging provides a way to obtain the malformed message. I guess it depends on how valuable the exception message is for debugging the json errors.

Comment (guyboertje) [inline lib/logstash/codecs/json_lines.rb]: i am removing the log line.

**Comment by jsvd** [comment]
LGTM

**Pull request merged**